cmd/install.go

```diff
@@ -76,117 +67,83 @@ func DownloadKubectl(version string) error {
+		fmt.Printf("%s is already installed.", version)
+		os.Exit(0)
 	}
 
 	// Create temp file of kubectl version in tmp directory
-	out, err := os.Create(homeDir + "/.kubemngr/kubectl-" + version)
+	out, err := os.Create(kubectl)
 	if err != nil {
 		log.Fatal(err)
 	}
-
 	defer out.Close()
 
-	// Get OS information to filter download type i.e linux / darwin
-	uname := GetOSInfo()
-
+	uname := getOSInfo()
 	// Compare system name to set value for building url to download kubectl binary
-	if arrayToString(uname.Sysname) == "Linux" {
-		sys = "linux"
-	} else if arrayToString(uname.Sysname) == "Darwin" {
-		sys = "darwin"
-	} else {
-		sys = "UNKNOWN"
-		fmt.Println("Unknown system")
+	if uname.Sysname != "Linux" && uname.Sysname != "Darwin" {
+		fmt.Printf("Unsupported OS: %s\nCheck github.com/zee-ahmed/kubemngr for issues.", uname.Sysname)
+		os.Exit(0)
+	}
+	if uname.Machine != "arm" && uname.Machine != "arm64" && uname.Machine != "x86_64" {
+		fmt.Printf("Unsupported arch: %s\nCheck github.com/zee-ahmed/kubemngr for issues.", uname.Machine)
+		os.Exit(0)
 	}
 
-	if arrayToString(uname.Machine) == "arm" {
-		machine = "arm"
-	} else if arrayToString(uname.Machine) == "arm64" {
-		machine = "arm64"
-	} else if arrayToString(uname.Machine) == "x86_64" {
+	var sys = strings.ToLower(uname.Sysname)
+	var machine string
+	if uname.Machine == "x86_64" {
 		machine = "amd64"
 	} else {
-		machine = "UNKNOWN"
-		fmt.Println("Unknown machine")
+		machine = strings.ToLower(uname.Machine)
 	}
 
-	url := "https://storage.googleapis.com/kubernetes-release/release/" + version + "/bin/" + sys + "/" + machine + "/kubectl"
-
-	resp, err := http.Get(url)
-	if err != nil {
-		log.Fatal(err)
+	// Check to make sure the file is a binary before moving the contents over to the user's home dir
+	url := "https://storage.googleapis.com/kubernetes-release/release/%v/bin/%v/%v/kubectl"
+	client := getter.Client{
+		Src:              fmt.Sprintf(url, version, sys, machine),
+		Dst:              kubectl,
+		ProgressListener: defaultProgressBar,
 	}
-
-	defer resp.Body.Close()
-
-	// Initialise WriteCounter and copy the contents of the response body to the tmp file
-	counter := &WriteCounter{}
-	_, err = io.Copy(out, io.TeeReader(resp.Body, counter))
+	fmt.Printf("Downloading %v\n", client.Src)
+	err = client.Get()
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	// The progress use the same line so print a new line once it's finished downloading
-	fmt.Println()
-
-	// Check to make sure the file is a binary before moving the contents over to the user's home dir
-	buf, _ := ioutil.ReadFile(homeDir + "/.kubemngr/kubectl-" + version)
-
 	// elf - application/x-executable check
-	if !filetype.IsArchive(buf) {
-		fmt.Println("failed to download kubectl file. Are you sure you specified the right version?")
-		os.Remove(homeDir + "/.kubemngr/kubectl-" + version)
+	mime, _, err := mimetype.DetectFile(kubectl)
+	if mime != "application/octet-stream" {
```

Comment:
This will only work with OSX or other platforms that have the output for `application/octet-stream`. Linux requires `application/x-executable`. 

We should replace this with: `if mime != "application/octet-stream" && mime != "application/x-executable"`

By doing that, the install/download for the kubectl binary will be successful on a linux machine.

Reply:
I'm unsure what Windows will read the file as but we should look into that.